Source/Core/Gen4/Generators/WildGenerator4.cpp

```diff
@@ -244,6 +244,160 @@ std::vector<WildState4> WildGenerator4::generateMethodJ(u32 seed, const Encounte
     return states;
 }
 
+template <bool bug>
+void createPokemon(PokeRNG &go, WildState4 &state, u8 buffer, u8 synchNature, u8 genderRatio, u16 tsv, const Lead &lead,
+                   const StateFilter &filter)
+{
+    u32 pid = 0;
+    if constexpr (bug)
+    {
+        bool cuteCharmSuccess = false;
+        if (lead == Lead::CuteCharmFemale || lead == Lead::CuteCharm25M || lead == Lead::CuteCharm50M || lead == Lead::CuteCharm75M
+            || lead == Lead::CuteCharm875M)
+        {
+            if ((go.nextUShort<true>() % 3) != 0) // Successfull cute charm
+            {
+                // Get hunt nature
+                state.setNature(go.nextUShort<true>() % 25);
+
+                if (!filter.compareNature(state))
+                {
+                    return;
+                }
+
+                cuteCharmSuccess = true;
+                pid = buffer + state.getNature();
```

Comment:
Add a "return true" statement here. Cute charm bypasses the for loop below.

Just make sure to add IVs here as well.

Reply:
It still has to run through one iteration of the loop to generate the IVs, nature, etc. We can have it either call createPokemon recursively but with the bool as false (but then it would check chute charm again) or just reintroduce the loops variable and change it to 1 instead of 4 in the case of successful cute charm.